api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/processor/RequestPaymentMessageProcessor.java

```diff
@@ -22,13 +39,49 @@
 @Slf4j
 public class RequestPaymentMessageProcessor implements MessageTypeProcessor {
 
+    private PaymentService paymentService;
+    private MessageContextLoader messageContextLoader;
+    private MessageQueueClient messageQueueClient;
+    private PaymentCycleService paymentCycleService;
+    private EventAuditor eventAuditor;
+
     @Override
     @Transactional(Transactional.TxType.REQUIRES_NEW)
     public MessageStatus processMessage(Message message) {
-        // TODO AFHS-405 add processing logic when implementing
+        RequestPaymentMessageContext messageContext = messageContextLoader.loadRequestPaymentMessageContext(message);
+        PaymentCycle paymentCycle = messageContext.getPaymentCycle();
+        PaymentType paymentType = messageContext.getPaymentType();
+
+        if (paymentType == PaymentType.FIRST_PAYMENT) {
+            PaymentCalculation paymentCalculation = aFullPaymentCalculation(paymentCycle.getTotalEntitlementAmountInPence());
+            makePayment(paymentCycle, INITIAL_PAYMENT, paymentCalculation, paymentType);
+        } else {
```

Comment:
Just checking here again, but does this equally apply for all the other `PaymentType` values (there's more than just 2 values in the enum)?

Reply:
It's only the case of first payments where we don't check the balance, or at least that's what we've done so far. If we think we shouldn't check for an interim payment then that should be it's own story as it's a change in behaviour.

Reply:
As far as I'm aware interim payments would also need the balance check. There's a story on the backlog somewhere to add a balance check override, which might be necessary once manual top-up payments are a thing.

Reply:
Can we add a TODO that references that story then to make sure that we keep that in context to these changes?